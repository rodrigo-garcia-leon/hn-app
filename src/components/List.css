.List ul {
  list-style-type: none;
}
