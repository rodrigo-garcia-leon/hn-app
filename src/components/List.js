import React from "react";
import ItemView from "./ItemView";
import "./List.css";

function List({ items }) {
  if (!items) {
    return null;
  }

  return (
    <section className="List">
      <ol>
        {items.map(item => (
          <li key={item}>
            <ItemView id={item}></ItemView>
          </li>
        ))}
      </ol>
    </section>
  );
}

export default List;
