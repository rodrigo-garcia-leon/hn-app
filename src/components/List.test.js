import React from "react";
import { create, act } from "react-test-renderer";
import List from "./List";
import {
  TEST_LIST,
  TEST_ITEM_22069310,
  TEST_ITEM_22089166,
  TEST_ITEM_22089546,
  setupFetchErrorMock
} from "../services/data.test";

describe("List", () => {
  let element, mockFetch;

  const createElement = async () =>
    act(async () => {
      element = create(<List items={TEST_LIST} />).toJSON();
    });

  beforeEach(() => {
    mockFetch = null;
  });

  afterEach(() => {
    if (mockFetch) {
      mockFetch.mockRestore();
    }
  });

  test("empty", () => {
    element = create(<List />).toJSON();

    expect(element).toMatchSnapshot();
  });

  test("network error", async () => {
    mockFetch = setupFetchErrorMock();

    await createElement();
    expect(element).toMatchSnapshot();
  });

  test("ok", async () => {
    mockFetch = jest
      .spyOn(global, "fetch")
      .mockResolvedValueOnce({
        json: jest.fn().mockResolvedValue(TEST_ITEM_22069310)
      })
      .mockResolvedValueOnce({
        json: jest.fn().mockResolvedValue(TEST_ITEM_22089166)
      })
      .mockResolvedValueOnce({
        json: jest.fn().mockResolvedValue(TEST_ITEM_22089546)
      });

    await createElement();
    expect(element).toMatchSnapshot();
  });
});
