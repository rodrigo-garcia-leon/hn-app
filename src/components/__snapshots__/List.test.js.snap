// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List empty 1`] = `null`;

exports[`List network error 1`] = `null`;

exports[`List ok 1`] = `null`;
